Watchman.Discord/Areas/Responses/Controllers/ResponsesController.cs

```diff
@@ -26,103 +27,87 @@ public ResponsesController(MessagesServiceFactory messagesServiceFactory, Servic
+            if (defaultResponse.Message == command.Message)
             {
                 await this._responsesService.RemoveResponse(response.OnEvent, response.ServerId);
-                await messageService.SendResponse(x => x.ResponseTheSameAsDefault(contexts, onEvent));
+                await messageService.SendResponse(x => x.ResponseTheSameAsDefault(contexts, command.OnEvent));
                 return;
             }
 
-            await this._responsesService.UpdateResponse(response.Id, message);
-            await messageService.SendResponse(x => x.ResponseHasBeenUpdated(contexts, onEvent, response.Message, message));
+            await this._responsesService.UpdateResponse(response.Id, command.Message);
+            await messageService.SendResponse(x => x.ResponseHasBeenUpdated(contexts, command.OnEvent, response.Message, command.Message));
         }
 
         [AdminCommand]
-        [DiscordCommand("remove response")]
-        public async Task RemoveResponse(DiscordRequest request, Contexts contexts)
+        public async Task RemoveResponse(RemoveResponseCommand command, Contexts contexts)
         {
             var messageService = this._messagesServiceFactory.Create(contexts);
-            var onEvent = request.Arguments.FirstOrDefault(x => x.Name?.ToLowerInvariant() == "onevent")?.Value;
-            if (onEvent == null)
-            {
-                await messageService.SendResponse(x => x.NotEnoughArguments());
-                return;
-            }
-            var response = await this._responsesService.GetResponseByOnEvent(onEvent, contexts.Server.Id);
+            var response = await this._responsesService.GetResponseByOnEvent(command.OnEvent, contexts.Server.Id);
             if (response == null)
             {
-                await messageService.SendResponse(x => x.ResponseNotFound(contexts, onEvent));
+                await messageService.SendResponse(x => x.ResponseNotFound(contexts, command.OnEvent));
                 return;
             }
-            await this._responsesService.RemoveResponse(onEvent, contexts.Server.Id);
-            await messageService.SendResponse(x => x.ResponseHasBeenRemoved(contexts, onEvent));
+            await this._responsesService.RemoveResponse(command.OnEvent, contexts.Server.Id);
+            await messageService.SendResponse(x => x.ResponseHasBeenRemoved(contexts, command.OnEvent));
         }
 
         [AdminCommand]
-        [DiscordCommand("responses")]
-        public async Task Responses(DiscordRequest request, Contexts contexts)
+        public async Task Responses(ResponsesCommand command, Contexts contexts)
         {
-            var argument = request.Arguments?.FirstOrDefault()?.Name?.ToLowerInvariant();
-            if (!this._possibleArguments.Contains(argument))
+            string typeOfResponses;
+            if (command.Default)
+            {
+                typeOfResponses = "default";
```

Comment:
ja bym tu zrobił na zasadzie nameof
typeOfResponses = nameof(command.Default);
i potem w tamtym switchu również porównywał po tych nameof

Jak coś się w kodzie zmieni to będzie błąd kompilacji, więc nie zostanie przeoczone coś w innym miejscu

Reply:
W sensie, że w tamtym switchu dać też nameof(command.Default) itd. ?

Reply:
tak, że np. `case nameof(command.Default)`
później jak dojdą enumy do IBotCommand to zmieni się ze stringa na enum

Reply:
to już chyba będzie lepiej jak zamiast przekazywać stringa do PrintResponses to dam po prostu command i wtedy ta metoda sobie sama sprawdzi. Będzie mniej sprawdzania i w dodatku nie będzie trzeba przekazywać 3 parametrów tylko dwa jak do tej pory.